### ISSUE TITLE
Problems with ".html"

### ISSUE THREAD

**Issue by JJ** [perl6/doc#1971]
## The problem

htmlify.p6 yields two errors, probably related to the same file:

```
html/syntax/.html got empty filename
```

**Comment by JJ** [comment]
It's generated by the '' page, or at least that is what appears in the '.html' file.

**Comment by ronaldxs** [comment]
If you go into `doc/Language/glossary.pod6` and remove the sections for `=head1 Sigil` and `=head1 Variable` then the two `got empty filename` errors go away.

In `htmlify.p6` routine `sub find-definitions(...)` there is a clause:

```perl6
given $subkinds {
...
    when 'variable'|'sigil'|'twigil'|'declarator'|'quote' {
        # TODO: More types of syntactic features
        %attr = :kind<syntax>,
                     :categories($subkinds),
        ;
    }
...
```
Most (all other?) entries from Language/glossary.pod6 are ignored but, because of the clause above, 'Sigil' and 'Variable' are added to the document registry `$*DR` with empty names.  Later, when `write-disambiguation-files` tries to loop through `for $*DR.grouped-by('name').kv -> $name, $p` the empty name is used for the `.html` file.

I don't understand enough about disambiguation and what `find-definitions` is trying to collect in `$*DR` to try to fix.  I looked at some of local `#` document lines in the local code but could not figure out enough from there and would be interested if someone knows of other helpful documentation.

**Comment by W4anD0eR96** [comment]
For the sigil part, I checked https://docs.perl6.org/js/search.js and lookup https://docs.perl6.org/language/variables.html#index-entry-%21%2525_%28sigil%29

All sigil are X<>-ed manually. I'd like to remove `'sigil'` for the very line, since its meaning is weird. If we want to make something searchable/indexed, use X<> as Pod6 spec.

**Comment by W4anD0eR96** [comment]
Oh, I just find it is about disambiguation...

**Comment by JJ** [comment]
This was fixed by #1997 . Thanks!
